### ISSUE TITLE
Assignment 1 Part B

### ISSUE THREAD

**Issue by progvasu** [goodboyanush/iiit-bangalore-march-april-2019#2]
Hi Anush, 

For converting a random noisy image to MNIST image is it necessary to start with (4, 4, 8) image or can we choose some other starting dimension.

Also a simple three layer decoder using (4, 4, 8) input is not able to reproduce anything even after 100 epochs of training. Can we use the concepts learnt in the last lecture to modify our decoder to something more powerful!

**Comment by vibhavagarwal5** [comment]
Yeah even I tried this and am getting an output like this (in the 2nd row). 
<img width="823" alt="Screenshot 2019-04-11 at 10 50 02 PM" src="https://user-images.githubusercontent.com/23319631/55977336-3253e500-5cac-11e9-8280-be7fb63b634b.png">

**Comment by progvasu** [comment]
Yes, getting the same output

**Comment by satu0king** [comment]
I am actually confused with what the Question is. Do we have to train an encoder-decoder just like the given sample code on MNIST dataset. Then take the decoder and try decoding random input data ? @goodboyanush

**Comment by progvasu** [comment]
The way I understood was to take random image's and to train just the decoder. Then if we supply any random image it should give a digit image like in MNIST dataset.

**Comment by adarshdas1995** [comment]
Getting the same outputs. the loss curve just plateaued for me. I tried lowering the learning rate but no change in output. Any other ideas ?

**Comment by progvasu** [comment]
I tried playing around and came with three different architectures by changing the input image dimensions and using different sets of layers with appx. 4500, 45000, 26000 parameters and interestingly or weirdly enough the loss plateau's at 0.2629 after 100 epochs of training for each one of them. Also the decoded images looks exactly the same for all the three.

**Comment by goodboyanush** [comment]
@progvasu you are right! you need to train a decoder from scratch. And feel free to use all the techniques discussed in the class room to achieve the same thing.

Tip: If the loss plateau's check for two things:
1. Either could be overfitting: try dropout, or different kind of regularizers
2. Could be that the model's capacity is poor. Try using more number of hidden layers with lesser number of filters in each layer.
